samcli/lib/test_runner/test_runner_template_generator.py

```diff
@@ -1,276 +1,214 @@
-    str:
-        The list of IAM statements in the form of a single YAML string corresponding to the resources specified by the given tags.
-
-        Returns a YAML comment error message if no resources match the tag, returns None if the tagging api call fails.
-
-    Raises
-    ------
-    KeyError:
-        If the specified tag does not match any resources.
-    """
-    resource_tag_mapping_list = _query_tagging_api(tag_filters, boto_client_provider)
-
-    if resource_tag_mapping_list is None:
-        raise KeyError(
-            "Got unexpected response from resource group tagging api. Response did not contain `ResourceTagMappingList` key."
+    def __init__(self, resource_arn_list: Optional[List[str]]):
+        self.resource_arn_list = resource_arn_list
+
+    def _get_jinja_template_string(self) -> str:
+        """
+        Returns the base jinja template string for the Test Runner CloudFormation Template
+        """
+        with open(self.TEST_RUNNER_JINJA_TEMPLATE_PATH) as test_runner_jinja_template:
+            return test_runner_jinja_template.read()
+
+    def _extract_api_id_from_arn(self, api_arn: str) -> str:
+        """
+        Extracts an api id from an HTTP or REST api arn.
+
+        NOTE: https://docs.aws.amazon.com/apigateway/latest/developerguide/arn-format-reference.html
+
+        REST API Arn: `arn:partition:apigateway:region::/restapis/api-id`
+
+        HTTP API Arn: `arn:partition:apigateway:region::/apis/api-id`
+
+        Parameters
+        ----------
+        api_arn : str
+            The arn of the HTTP or REST api from which the api id is extracted.
+
+        Returns
+        -------
+        str
+            The api id from the api arn.
+        """
+
+        return api_arn[api_arn.rindex("/") + 1 :]
+
+    def _create_iam_statement_string(self, resource_arn: str) -> Union[dict, None]:
+        """
+        Returns an IAM Statement in the form of a YAML string corresponding to the supplied resource ARN.
+
+        The generated Actions are commented out to establish a barrier of consent, preventing customers from accidentally granting permissions that they did not mean to.
+
+        The Action list is empty if there are no IAM Action permissions to generate for the given resource.
+
+        Returns `None` if the `resource_arn` corresponds to an IAM or STS resource.
+
+        Parameters
+        ----------
+        resource_arn : str
+            The arn of the resource for which the IAM statement is generated.
+
+        Returns
+        -------
+        str:
+            An IAM Statement in the form of a YAML string.
+
+            `None` if the `resource_arn` corresponds to an IAM or STS resource.
+
+        Raises
+        ------
+        jinja2.exceptions.TemplateError:
+            If the IAM Statement template render fails.
+        """
+
+        partition_regex = r"(aws|aws-cn|aws-us-gov)"
+        region_regex = r"(us(-gov)?|ap|ca|cn|eu|sa)-(central|(north|south)?(east|west)?)-\d"
```

Comment:
Any reason not to use `[\w-]+` for both? Having to maintain a list of current regions seems painful.

Reply:
Yeah, I see. I wanted to keep the regex strict for clarity sake, but we don't want to take up the burden of maintaining the regions.